src/stack.h

```diff
@@ -0,0 +1,81 @@
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef DEBUG
+    #define LOG(args...) fprintf(stderr, args);
+#else
+    #define LOG(args...)
+#endif
+
+typedef struct ScannerStack {
+  unsigned int stack[256];
```

Comment:
This stack does not re-allocate if it reaches the limit and I've chosen a rather optimistic 256 levels of indentation as being absolute maximum that a sane person would use..